terraform/scripts/create_bootstrap_bundle.sh

```diff
@@ -0,0 +1,181 @@
+log "Version ${hab_version}"
+
+########################################################################
+# Preliminaries, Helpers, Constants
+
+find_if_exists() {
+    command -v ${1} || { log "Required utility '${1}' cannot be found!  Aborting."; exit 1; }
+}
+
+# These are the key utilities this script uses. If any are not present
+# on the machine, the script will exit.
+aws=$(find_if_exists aws)
+awk=$(find_if_exists awk)
+tar=$(find_if_exists tar)
+hab=$(find_if_exists hab)
+shasum=$(find_if_exists shasum)
+sort=$(find_if_exists sort)
+
+# hab-launcher is versioned differently than the other packages
+# (monotonically increasing integer only). It also isn't going to
+# change much at all. This is the current version.
+#
+# TODO: Accept this as an optional argument?
+launcher_version=4435
+
+# The packages needed to run a Habitat Supervisor. These will be
+# installed on all machines.
+sup_packages=(core/hab-launcher/${launcher_version}
+              core/hab/${hab_version}
+              core/hab-sup/${hab_version}
+              core/hab-butterfly/${hab_version})
+
+# If the HAB_DEPOT_URL environment variable is set, we'll use that
+# when downloading packages. Otherwise, we'll just default to the
+# production depot.
+if [[ ! -z "${HAB_DEPOT_URL:-}" ]]
+then
+    log "Using HAB_DEPOT_URL from environment: ${HAB_DEPOT_URL}"
+    depot_flag="HAB_DEPOT_URL=${HAB_DEPOT_URL}"
+else
+    log "No HAB_DEPOT_URL detected; using the default"
+    depot_flag=
+fi
+
+# All packages that compose the Builder / Depot service. Not all need
+# to be installed on the same machine, but all need to be present in
+# our bundle.
+builder_packages=(core/builder-api
+                  core/builder-api-proxy
+                  core/builder-admin
+                  core/builder-admin-proxy
+                  core/builder-datastore
+                  core/builder-jobsrv
+                  core/builder-originsrv
+                  core/builder-router
+                  core/builder-scheduler
+                  core/builder-sessionsrv
+                  core/builder-worker)
+
+# This is where we ultimately put all the things in S3.
+s3_bucket="habitat-builder-bootstrap"
+
+# This is the name by which we can refer to the bundle we're making
+# right now. Note that other bundles can be made that contain the
+# exact same packages.
+this_bootstrap_bundle=hab_builder_bootstrap_$(date +%Y%m%d%H%M%S)
+
+########################################################################
+# Download all files locally
+
+# Because Habitat may have already run on this system, we'll want to
+# make sure we start in a pristine environment. That way, we can just
+# blindly copy everything in ${sandbox_dir}/hab/cache/artifacts, confident
+# that those artifacts are everything we need, and no more.
+sandbox_dir=${this_bootstrap_bundle}
+mkdir ${sandbox_dir}
+log "Using ${sandbox_dir} as the Habitat root directory"
+
+for package in "${sup_packages[@]}" "${builder_packages[@]}"
```

Comment:
If we end up running this "raw" on a system we'll want to make sure we have Bash 4.x+ (IIRC) to get the array behavior. I don't think this'll be an issue, but that's come up with our build program in the past when running out of Studio.